### PR TITLE
fix(ngMock): Change ErrorAddingDeclarationLocationStack prototype

### PR DESCRIPTION
**Pull request by peabnuts123**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
fix to angular-mocks for ErrorAddingDeclarationLocationStack error to change its prototype to Error.prototype

**What is the current behavior? (You can also link to an open issue here)**
Testing frameworks and other libraries interacting with angular cannot correctly detect this error with `instanceof`

**What is the new behavior (if this is a feature change)?**
This Error is correctly inherited from `Error`. `ErrorAddingDeclarationLocationStack instanceof Error === true`.

**Does this PR introduce a breaking change?**
No, in fact it only changes one line.

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/angular/angular.js/blob/master/CONTRIBUTING.md#commit-message-format
- [X] Tests for the changes have been added (for bug fixes / features)

**Other information**:
Apologies for ridiculous branch name. I hope this was all done correctly, I am still new to git/GitHub :octocat: 

change prototype to Error.prototype so test frameworks can catch it properly
#13821

### CODE REVIEW

**Comment by peabnuts123** [comment]
Fix for issue [#13821](https://github.com/angular/angular.js/issues/13821)

File: test/ngMock/angular-mocksSpec.js [hunk 1]
```diff
@@ -2505,6 +2505,44 @@ describe('`afterEach` clean-up', function() {
...
+        });
+
+        //call the injector on 'causeError' to make sure the provider is called
+        var causeError;
+        inject(function(_causeError_) {
+          causeError = _causeError_;
+        });
+      }
+
+      //The following is a simplified implementation of future Jasmine's .toThrowError()
```
Comment (gkalpak) [inline test/ngMock/angular-mocksSpec.js]: What do you mean by `future Jasmine's .toThrowError()` ?
Why can't you use it directly ?

Comment (peabnuts123) [inline test/ngMock/angular-mocksSpec.js]: `toThrowError()` was introduced in Jasmine 2.0. It uses `errorObject instanceof Error` to check if the thrown object is an Error Object (for matching strings/regex against their `.message` property). Angular's build environment uses now >2-year-old Jasmine 1.3 for its own tests, so I had to rig a basic function for how Jasmine's `toThrowError()` fails when running this code.

Comment (gkalpak) [inline test/ngMock/angular-mocksSpec.js]: Not any more :) We are on Jasmine 2 now :sunglasses:

Comment (peabnuts123) [inline test/ngMock/angular-mocksSpec.js]: Whoo! Let me amend this PR in the coming days

**Comment by Narretz** [comment]
@peabnuts123 Did you ever get around to updating this PR?

**Comment by peabnuts123** [comment]
Unfortunately I've been pretty busy and have been unable to update this. I still intend to, as this fix is something I need in my project. Currently just using my own fork in our build...

**Comment by peabnuts123** [comment]
I've changed the unit test to utilise the real Jasmine `toThrowError` as mentioned in the previous comments. Apologies it took so long. All the builds work on my end, and I've confirmed everything works against `master` branch AND is still a necessary change. 

The Travis build appears to have failed unrelated to anything I've changed? Can this be confirmed / re-run?

**Comment by gkalpak** [comment]
Travis is green now :+1:

**Pull request closed without merging**